### PR TITLE
stm32: Get STM32_LPTIM_CLOCK symbol defined from device tree

### PR DESCRIPTION
**Pull request by erwango**
Simplify LPTIM related configuration by defining `STM32_LPTIM_CLOCK` choice symbol defined from device tree description

Fixes following warning when `CONFIG_PM=n`:
```
warning: The choice symbol STM32_LPTIM_CLOCK_LSE (defined at drivers/timer/Kconfig.stm32_lptim:30)
was selected (set =y), but no symbol ended up as the choice selection. See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_STM32_LPTIM_CLOCK_LSE and/or look up
STM32_LPTIM_CLOCK_LSE in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.
```

### CODE REVIEW

File: scripts/kconfig/kconfigfunctions.py [hunk 1]
```diff
@@ -547,18 +574,46 @@ def dt_node_array_prop(kconf, name, path, prop, index, unit=None):
         'g' or 'G'  divide by 1,073,741,824 (1 << 30)
     """
     if doc_mode or edt is None:
-        return "0"
+        return "9"
```
Comment (erwango) [inline scripts/kconfig/kconfigfunctions.py:577]: Leftover

File: scripts/kconfig/kconfigfunctions.py [hunk 2]
```diff
@@ -547,18 +574,46 @@ def dt_node_array_prop(kconf, name, path, prop, index, unit=None):
...
 
     try:
         node = edt.get_node(path)
     except edtlib.EDTError:
-        return "0"
+        return "8"
```
Comment (erwango) [inline scripts/kconfig/kconfigfunctions.py:582]: Leftover

**Review by FRASTM — DISMISSED** [review]

File: scripts/kconfig/kconfigfunctions.py [hunk 3]
```diff
@@ -228,6 +228,32 @@ def _node_array_prop(node, prop, index=0, unit=None):
...
+    """
+    if not node:
+        return 1
+
+    if prop not in node.props:
+        return 2
+    if node.props[prop].type != "array":
+        return 3
+    if int(index) >= len(node.props[prop].val):
+        return 4
```
Comment (gautierg-st) [inline scripts/kconfig/kconfigfunctions.py:245]: Should the function really return these values? Or is this a leftover from when you tested your changes?

File: include/zephyr/dt-bindings/clock/stm32_common_clocks.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32_COMMON_CLOCKS_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_CLOCK_STM32_COMMON_CLOCKS_H_
+
+/** System clock */
+#define STM32_SRC_SYSCLK	0x001
+/** Fixed clocks  */
+#define STM32_SRC_LSE		0x002
+#define STM32_SRC_LSI		0x003
+
+/** Dummy: Add a specificier when no selection is possible */
```
Comment (gautierg-st) [inline include/zephyr/dt-bindings/clock/stm32_common_clocks.h:15]: Specifier maybe?

**Review by gautierg-st — CHANGES_REQUESTED** [review]

File: scripts/kconfig/kconfigfunctions.py (hunk 3 above)
Comment (erwango) [reply scripts/kconfig/kconfigfunctions.py:245]: Good catch. Leftover indeed

**Review by gautierg-st — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**